Refactor experimental ML CLI and code (#1218)
elastic/detection-rules: docs/experimental-machine-learning/DGA.md

# Machine Learning on Domain Generation Algorithm (DGA)

> Several blogs were put out on how you can create and leverage supervised DGA ML models to enrich data within the stack.

Reviewer: awkward wording. "were put out on how you can"
```suggestion
To create and use supervised DGA ML models to enrich data within the stack, check out these Elastic blogs:
```

  -eu, --es-user TEXT
  --cloud-id TEXT
  --elasticsearch-url TEXT

* experimental commands are use at your own risk and may change without warning *

Usage: detection_rules es experimental ml setup [OPTIONS]

  Upload ML model and dependencies to enrich data.

Options:
  -t, --model-tag TEXT       Release tag for model files staged in detection-
                             rules (required to download files)
  -r, --repo TEXT            GitHub repository hosting the model file releases
                             (owner/repo)
  -d, --model-dir DIRECTORY  Directory containing local model files
  --overwrite                Overwrite all files if already in the stack
  -h, --help                 Show this message and exit.

```

### Detailed steps

#### 1. Upload and setup the model file and dependencies

Run `python -m detection_rules es <args_or_config> experimental ml setup -t <release-tag>`

*If updating a new model, you should first uninstall any existing models using `remove-model`*

You can also upload files locally using the `-d` option, so long as the naming convention of the files match the 
expected pattern for the filenames.

#### 2. Update packetbeat configuration

> You will need to update your packebeat.yml config file to point to the enrichment pipeline

Reviewer: What if I'm using Fleet?
```suggestion
You will need to update your packeybeat.yml config file to point to the enrichment pipeline
```

Reviewer 2: Updated.

Author: if fleet, then you should be able to just configure the ingest pipeline in the elasticsearch index config, IINM
